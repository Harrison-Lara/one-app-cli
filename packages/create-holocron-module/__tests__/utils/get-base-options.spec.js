const prompts = require('prompts');
const getBaseOptions = require('../../src/utils/get-base-options');

jest.mock('prompts', () => jest.fn(() => 'promptsResponseMock'));

describe('getBaseOptions', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('should call prompts with the correct set of base options', () => {
    getBaseOptions();
    expect(prompts).toHaveBeenCalledTimes(1);
    // snapshot params as its a large array that will grow over time.
    expect(prompts.mock.calls[0]).toMatchSnapshot();
  });
});
