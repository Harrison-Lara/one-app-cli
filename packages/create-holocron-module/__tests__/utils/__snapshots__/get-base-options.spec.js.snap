// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBaseOptions should call prompts with the correct set of base options 1`] = `
Array [
  Array [
    Object {
      "initial": "my-module",
      "message": "Enter module name:",
      "name": "moduleName",
      "type": "text",
    },
  ],
]
`;
